### PR TITLE
Removing color hack + preventing filter changes every frame

### PR DESCRIPTION
**Pull request by gamedevsam**
We need to ensure this doesn't cause massive breakages, but I support this change for a major patch.

### CODE REVIEW

File: flixel/FlxCamera.hx [hunk 1]
```diff
@@ -1256,10 +1255,8 @@ class FlxCamera extends FlxBasic
 			{
 				return;
 			}
-			// This is temporal fix for camera's color
```
Comment (Gama11) [inline flixel/FlxCamera.hx]: What was this even for?

**Comment by Gama11** [comment]
What's a "major patch"? SemVer is `Major.Minor.Patch`, "major patch" is an oxymoron. :D

What's the benefit of this change? Does the setter cause allocations you can see in hxscout?

File: flixel/FlxCamera.hx (hunk 1 above)
Comment (gamedevsam) [inline flixel/FlxCamera.hx]: I don't know, seemed very hacky to me, so I removed it. As far as I can tell everything is running fine.

**Comment by gamedevsam** [comment]
Yes, the set causes a lot of Array allocations on the OpenFL DisplayObject.

http://screencast.com/t/mdy5AXue

File: flixel/FlxCamera.hx (hunk 1 above)
Comment (Gama11) [inline flixel/FlxCamera.hx]: Wow, this code was _old_! https://github.com/HaxeFlixel/flixel/commit/227a4647b8b1f38b11eacc98bf1be57203c1381e#diff-9ef1aac582dab97c9bde1a8fb693a1b7R845

**Pull request merged**